satpy/tests/reader_tests/test_hsaf_h5.py

```diff
@@ -0,0 +1,77 @@
+"""Tests for the H-SAF H5 reader."""
+import os
+import tempfile
+from datetime import datetime
+
+import h5py
+import numpy as np
+
+from satpy import Scene
+from satpy.resample import get_area_def
+
+# real shape is 916, 1902
+shape_sc = (916, 1902)
+shape_sc_colormap = (256, 3)
+
+
+def hsaf_filename():
+    """Create a fake HSAF SC HDF5 file."""
+    dirname = tempfile.mkdtemp()
+    filename = dirname + "/h10_20221115_day_merged.H5"
+    with h5py.File(filename, mode="w") as h5f:
+        h5f.create_dataset('SC', shape_sc, dtype=np.uint8)
+        h5f.create_dataset('colormap', shape_sc_colormap, dtype=np.uint8)
+    return filename
+
+
+def test_hsaf_sc_dataset():
+    """Test the H-SAF SC dataset."""
+    filename = hsaf_filename()
+
+    try:
+        scn = Scene(filenames=[str(filename)], reader="hsaf_h5")
+        scn.load(['SC'])
+    finally:
+        os.remove(filename)
```

Comment:
This will leave the temporary directory behind. I think `shutil.rmtree(os.path.dirname(hsaf_filename))` would be one option. Or a [context manager](https://book.pythontips.com/en/latest/context_managers.html#implementing-a-context-manager-as-a-class) that in `__exit__()` removes the temp directory. Or some readily existing fixture that I'm unaware of...

With the context manager the test could be something like
```python
    with HSAFFilename as fname:
        scn = Scene(...)
```
and the `__exit__()` method would then call `shutil.rmtree()` for the `dirname`.